src/Microsoft.AspNet.OData.Shared/ODataEdmAPIHandlerFactory.cs

```diff
@@ -0,0 +1,49 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataEdmAPIHandlerFactory.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Factory class for ODataAPIHandlers for typeless entities
+    /// </summary>
+    internal abstract class ODataEdmAPIHandlerFactory
+    {
+        protected ODataEdmAPIHandlerFactory(IEdmModel model)
+        {
+            Model = model;
+        }
+
+        /// <summary>
+        /// The IEdmModel for the Factory.
+        /// </summary>
+        public IEdmModel Model { get; private set; }
+
+        /// <summary>
+        /// Get the handler depending on OData path.
+        /// </summary>
+        /// <param name="odataPath">OData path corresponding to an odataid.</param>
```

Comment:
This being public documentation for this method, would most people be able to relate with "odataid"?